src/sub.c

```diff
@@ -1124,6 +1124,54 @@ natsSubscription_QueuedMsgs(natsSubscription *sub, uint64_t *queuedMsgs)
     return s;
 }
 
+int64_t
+natsSubscription_GetID(natsSubscription* sub)
+{
+    if (sub == NULL)
+        return 0;
+
+    natsSub_Lock(sub);
+
+    if (sub->closed)
+    {
+        natsSub_Unlock(sub);
+        return 0;
+    }
+
+    int64_t id = 0;
```

Comment:
I prefer that we declare this at the top of the function (to be consistent with rest of the code).